src/dune_lang/action.ml

```diff
@@ -493,7 +507,13 @@ let rec map_string_with_vars t ~f =
   | Diff diff -> Diff (Diff.map diff ~path:f ~target:f)
   | No_infer t -> No_infer (map_string_with_vars t ~f)
   | Pipe (o, ts) -> Pipe (o, List.map ts ~f:(map_string_with_vars ~f))
-  | Cram sw -> Cram (f sw)
+  | Cram { script; shell_spec } ->
+    let shell_spec =
```

Comment:
Let's put this in `Shell_spec` and call it `map`. The mli should mention how it acts on values.

Reply:
addressed in 179f7daa7